backend/external_api_handler.py

```diff
@@ -6,19 +6,41 @@
 
 def search_api(q_string):
     """Searches the MusixMatch API for songs and returns payload"""
-    r = requests.get(f'{API_BASE_URL}/track.search?q={q_string}&f_has_lyrics=1&page_size=100&s_artist_rating=desc&apikey={MUSIX_MATCH_API_KEY}')
-    data = r.json()
-    results = [{
-        "mmId": s["track"]["track_id"],
-        "title": s["track"]["track_name"],
-        "artist": s["track"]["artist_name"],
-        "lyricsLocation": s["track"]["track_share_url"].split('?', 1)[0],
-    } for s in data['message']['body']['track_list']]
+    try:
+        r = requests.get(f'{API_BASE_URL}/track.search?q={q_string}&f_has_lyrics=1&page_size=100&s_artist_rating=desc&apikey={MUSIX_MATCH_API_KEY}')
+        
+        if r.status_code == 200:
+            data = r.json()
+            results = [{
+                "mmId": s["track"]["track_id"],
+                "title": s["track"]["track_name"],
+                "artist": s["track"]["artist_name"],
+                "lyricsLocation": s["track"]["track_share_url"].split('?', 1)[0],
+            } for s in data['message']['body']['track_list']]
 
-    return results
+            return results
+        
+        if r.status_code == 404:
+            return {"message":"No results found"}
```

Comment:
Should raise something like `NotFoundError` instead of forcing the user of this function to inspect the return contents to know whether the search succeeded.